packages/@aws-cdk/cloudformation-include/lib/file-utils.ts

```diff
@@ -37,23 +37,26 @@ const shortForms: yaml_types.Schema.CustomTag[] = [
   makeTagForCfnIntrinsic('Ref', false),
   makeTagForCfnIntrinsic('Condition', false),
   makeTagForCfnIntrinsic('GetAtt', true, (_doc: yaml.Document, cstNode: yaml_cst.CST.Node): any => {
-    // The position of the leftmost period and opening bracket tell us what syntax is being used
-    // If no brackets are found, then the dot notation is being used; the leftmost dot separates the
-    // logical ID from the attribute.
-    //
-    // If a bracket is found, then the list notation is being used; if present, the leftmost dot separates the
-    // logical ID from the attribute.
-    const firstDot = cstNode.toString().indexOf('.');
-    const firstBracket = cstNode.toString().indexOf('[');
+    const parsedArguments = parseYamlStrWithCfnTags(cstNode.toString().substring('!GetAtt'.length));
 
-    return {
-      'Fn::GetAtt': firstDot !== -1 && firstBracket === -1
-        ? [
-          cstNode.toString().substring('!GetAtt '.length, firstDot),
-          parseYamlStrWithCfnTags((cstNode.toString().substring(firstDot + 1))),
-        ]
-        : parseYamlStrWithCfnTags(cstNode.toString().substring('!GetAtt'.length)),
-    };
+    let value: any;
+    if (typeof parsedArguments === 'string') {
+      // if the arguments to Fn::GetAtt are a string,
+      // the part before the first '.' is the logical ID,
+      // and the rest is the attribute name
+      // (which can contain '.')
+      const firstDot = parsedArguments.indexOf('.');
```

Comment:
The return of this is never checked. If the value is malformed (no dot), this means the first value (logical ID) will be blank and second value will contain the entire string. How (and when) would this error manifest to the user? Would it be worth an explicit check here to validate?

Reply:
Sure. Added.